src/ngraph/op/extractimagepatches.cpp

```diff
@@ -0,0 +1,157 @@
+
+#include "ngraph/op/extractimagepatches.hpp"
+#include "ngraph/attribute_visitor.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+// ExtractImagePatches v3
+
+constexpr NodeTypeInfo op::v3::ExtractImagePatches::type_info;
+
+op::v3::ExtractImagePatches::ExtractImagePatches(const Output<Node>& image,
+                                                 const Shape& sizes,
+                                                 const Strides& strides,
+                                                 const Shape& rates,
+                                                 const PadType& padding)
+    : Op({image})
+    , m_patch_sizes(sizes)
+    , m_patch_movement_strides(strides)
+    , m_patch_selection_rates(rates)
+    , m_padding(padding)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v3::ExtractImagePatches::validate_and_infer_types()
+{
+    const PartialShape input_Pshape = get_input_partial_shape(0);
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(0).is_integral_number(),
+                          "input tensor must be an integral number.");
+    NODE_VALIDATION_CHECK(this, input_Pshape.rank() == 4, "input tensor must be 4D tensor.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_sizes.size() == 2,
+                          "Attribute sizes should be in [size_rows, size_cols] format.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_movement_strides.size() == 2,
+                          "Attribute strides should be in [stride_rows, stride_cols] format.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_movement_strides[0] > 0 && m_patch_movement_strides[1] > 0,
+                          "Attribute strides should be strictly greater than zeros in values.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_selection_rates.size() == 2,
+                          "Attribute rates should be in [rate_rows, rate_cols] format.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_selection_rates[0] > 0 && m_patch_selection_rates[1] > 0,
+                          "Attribute rates should be strictly greater than zeros in values.");
+
+    NODE_VALIDATION_CHECK(
+        this,
+        m_padding == PadType::VALID || m_padding == PadType::SAME_LOWER ||
+            m_padding == PadType::SAME_UPPER,
+        "Attribute padding should be in either valid or same_lower or same_upper.");
+
+    if (input_Pshape.is_dynamic())
+    {
+        set_input_is_relevant_to_shape(0);
+        auto output_Pshape = PartialShape::dynamic(4);
+        set_output_type(0, get_input_element_type(0), output_Pshape);
+    }
+    else
+    {
+        Shape input_shape = get_input_shape(0); // as input shape is static
+        size_t out_rows(0);
+        size_t out_cols(0);
+
+        if (m_patch_movement_strides[0] > input_shape[2] ||
```

Comment:
Is this case really possible? The same comment for line 102. If they are possible then we need to have a unit test for each of them.

Reply:
These two "corner" cases are suggested by Scott. Actual questions were:
1. what happen if stride length is larger than data dimensions
2. what will happen if data is blank.

However, you are right, it looks like I don't need to have special case for these. (explain in the reply of your next comment.

I fixed the code calculations for large stride but smaller sizes.
According to your suggestion, I have added more test cases.